builder/azure/chroot/step_mount_device.go

```diff
@@ -0,0 +1,132 @@
+package chroot
+
+// mostly borrowed from ./builder/amazon/chroot/step_mount_device.go
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"github.com/hashicorp/packer/builder/amazon/chroot"
+	"github.com/hashicorp/packer/helper/multistep"
+	"github.com/hashicorp/packer/packer"
+	"github.com/hashicorp/packer/template/interpolate"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+var _ multistep.Step = &StepMountDevice{}
+
+type StepMountDevice struct {
+	MountOptions   []string
+	MountPartition string
+	MountPath      string
+
+	mountPath string
+}
+
+func (s *StepMountDevice) Run(ctx context.Context, state multistep.StateBag) multistep.StepAction {
+	ui := state.Get("ui").(packer.Ui)
+	device := state.Get("device").(string)
+	config := state.Get("config").(*Config)
+	wrappedCommand := state.Get("wrappedCommand").(chroot.CommandWrapper)
+
+	ictx := config.ctx
+
+	ictx.Data = &struct{ Device string }{Device: filepath.Base(device)}
+	mountPath, err := interpolate.Render(s.MountPath, &ictx)
+
+	if err != nil {
+		err := fmt.Errorf("error preparing mount directory: %s", err)
```

Comment:
NIT: Is %s intentional for err - instead of, say, %v? Also other places in this file

Reply:
From the [fmt package docs](https://golang.org/pkg/fmt/):

> Regardless of the verb, if an operand is an interface value, the internal concrete value is used, not the interface itself.
> ...
> If an operand implements the error interface, the Error method will be invoked to convert the object to a string, which will then be formatted as required by the verb (if any).

Since `err` is an `error` interface, it doesn't really matter if it's formatted with `%s` or `%v`. I have been using `%v` a lot lately in logging because it prints the object and doesn't complain about the wrong specifier.